### DISCUSSION TITLE
how to use useQuery to request many request

### DISCUSSION THREAD

**Discussion by vaynevayne** [TanStack/query#5013] [Q&A]
```jsx
  const apiGetData2 = useQuery({
    queryKey: ['/report_data/table', data],
    queryFn: () => axios({}),
  })

  const apiGetData = useCallback(
    (data: any) =>
      request({
        url: '/report_data/table',
        method: 'POST',
        data: data,
      }),
    [request]
  )
const fn = ()=> {
const res = await Promise.all([
apiGetData('1'),
apiGetData('2'),
apiGetData('3'),
])
}



```
<img width="579" alt="image" src="https://user-images.githubusercontent.com/124666577/220308435-bd0a390b-1e61-4d28-8c3a-e8a8e0480be3.png">

useQuery needs to write queryKey at the top level, but my request body is dynamically generated, it changes, and there will be a lot of requests, or even an indefinite number of requests. In this case, can't react-query be used?

**Comment by sapter** [comment]
Use [dependent queries](https://tanstack.com/query/v4/docs/react/guides/dependent-queries) where the key array of the dependent query contains the results of the query it depends on.
